merkle-tree/hash-set/src/lib.rs

```diff
@@ -431,60 +432,102 @@ where
         value: &BigUint,
         current_sequence_number: usize,
     ) -> Result<(), HashSetError> {
+        let index_bucket = self.find_element_iter(value, current_sequence_number, 0, 20)?;
+        let (value_index, is_new) = match index_bucket {
+            Some(value_index) => value_index,
+            None => {
+                return Err(HashSetError::Full);
+            }
+        };
+
+        match is_new {
```

Comment:
Nice. It was definitely a huge mistake from me to not differentiating these cases. Thanks for catching that! :+1: